hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/JmsSourceBuilder.java

```diff
@@ -147,13 +119,53 @@ public JmsSourceBuilder consumerFn(
     }
 
     /**
-     * Sets the function which commits the session after consuming each message.
+     * Sets the maximum processing guarantee for the source. You can use it to
+     * decrease the guarantee of this source compared to the job's guarantee.
+     * If you configure stronger guarantee than the job has, the job's
+     * guarantee will be used. Use it if you want to avoid the overhead of
+     * acknowledging the messages in transactions if you don't need it.
+     * <p>
+     * XA transactions are required for exactly-once mode. If your job uses
+     * exactly-once guarantee and your JMS client doesn't support XA
+     * transactions, use {@code AT_LEAST_ONCE} guarantee to use normal
+     * transactions. In this mode the transaction will be committed in the 2nd
+     * phase of the snapshot so you're guaranteed that each message will be
+     * processed at least once if the job is forced to restart.
      * <p>
-     * If not provided, the builder creates a no-op consumer.
+     * If you use {@link ProcessingGuarantee#NONE}, messages will be consumed
+     * in auto-acknowledge mode. In this mode some messages can be processed
+     * more than once and some not at all if the job is forced to restart.
+     * <p>
+     * The default is {@link ProcessingGuarantee#EXACTLY_ONCE}, which means
+     * that source's guarantee will match job's guarantee.
+     *
+     * @return this instance for fluent API
      */
-    public JmsSourceBuilder flushFn(@Nonnull ConsumerEx<? super Session> flushFn) {
-        checkSerializable(flushFn, "flushFn");
-        this.flushFn = flushFn;
+    public JmsSourceBuilder maxGuarantee(ProcessingGuarantee guarantee) {
```

Comment:
I would suggest to use different name - something like `decreasedGuarantee` or just `sourceGuarantee` sounds more intuitive for me.